File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Valid_Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s=="") return true;
        string st="";
        for(int i=0;i<s.size();i++){
            if((s[i]>='a' && s[i]<='z') || (s[i]>='0' && s[i]<='9')){
                st+=s[i];
            }
            else if(s[i]>='A' && s[i]<='Z'){
                st+=(s[i]+32);
            }
        }
        int i=0,j=st.size()-1;
        while(i<j){
            if(st[i]==st[j]){
                i++;
                j--;
            }
            else return false;
        }
        return true;
    }
};

File: Anushka-Pachauri/TwoPointer/ValidPalindrome.java
class ValidPalindrome {
    public boolean isPalindrome(String s) {
        if(s.isEmpty())
        return true;
        int start=0;
        int end=s.length()-1;
        while(start<=end)
        {
            char charstart=s.charAt(start);
            char charend=s.charAt(end);
            if(!Character.isLetterOrDigit(s.charAt(start)))
            start++;
            else if(!Character.isLetterOrDigit(s.charAt(end)))
            end--;
            else
            {
                if(Character.toLowerCase(charstart)!=Character.toLowerCase(charend))
                return false;
                start++;
                end--;
            }
        }
        return true;
    }
}